core/probePlugin/src/main/scala/org/virtuslab/ideprobe/handlers/ExpandMacro.scala

```diff
@@ -30,7 +31,7 @@ object ExpandMacro {
     val data = new THashMap[String, Object]()
     data.put(CommonDataKeys.PROJECT.getName, fileRef.project)
     data.put(CommonDataKeys.VIRTUAL_FILE.getName, file)
-    data.put(PlatformDataKeys.PROJECT_FILE_DIRECTORY.getName, project.getBaseDir)
+    data.put(PlatformDataKeys.PROJECT_FILE_DIRECTORY.getName, ProjectUtil.guessProjectDir(project))
```

Comment:
Huh is this equivalent to `getBaseDir`? 🤔

Reply:
I suppose that was what the deprecation warning pointed to?

Reply:
It should be, if I get the javadoc in source code right...
```
/**
 *  Tries to guess the "main project directory" of the project.
 *
 *  There is no strict definition of what is a project directory, since a project can contain multiple modules located in different places,
 *  and the `.idea` directory can be located elsewhere (making the popular [Project.getBaseDir] method not applicable to get the "project
 *  directory"). This method should be preferred, although it can't provide perfect accuracy either.
 */
fun Project.guessProjectDir() : VirtualFile? {
  if (isDefault) {
    return null
  }

  val modules = ModuleManager.getInstance(this).modules
  val module = if (modules.size == 1) modules.first() else modules.firstOrNull { it.name == this.name }
  module?.guessModuleDir()?.let { return it }
  return LocalFileSystem.getInstance().findFileByPath(basePath!!)
}
```

Reply:
Actually, it's [KDoc](https://kotlinlang.org/docs/kotlin-doc.html) in this case ;)

Reply:
But javadoc/KDoc/scaladoc are barely distinguishable from each other, save for what language they're used in